### PR TITLE
default should populate before save completes #28

### PR DESCRIPTION
**Pull request by seancookr**
@javierg @rubenrails @neojin @netlogics Review please

### CODE REVIEW

File: test/document_test.rb [hunk 1]
```diff
@@ -22,6 +22,10 @@ def add_to_foo key, value
   end
 end
 
+class TestFoo < Dolly::Document
+  property :default_test_propery, class_name: String, default: 'FOO'
```
Comment (rubenrails) [inline test/document_test.rb]: typo `default_test_property`

**Comment by rubenrails** [comment]
Other than comment above. :shipit:

File: test/document_test.rb (hunk 1 above)
Comment (seancookr) [inline test/document_test.rb]: hehe whoops :)

**Comment by neojin** [comment]
lol someone made a SaaS with rubocop!

**Comment by netlogics** [comment]
The double quote warnings are kind of annoying. Looks like we have some control over that.
```
Style/StringLiterals:
  Description: Checks if uses of quotes match the configured preference.
  StyleGuide: https://github.com/bbatsov/ruby-style-guide#consistent-string-literals
  Enabled: true
  EnforcedStyle: double_quotes
  SupportedStyles:
  - single_quotes
  - double_quotes
```

**Comment by seancookr** [comment]
back to just the failing tests.  I wrote another test to double check something and found a significant bug with how we define the setter methods.

    def write_property name, value
      @doc[name.to_s] = self.properties[name].value = value
    end

the problem is this changes the value of the properties in the class attribute, which is not good if you are going to init a new instance of the class, as the default value will be the value from the previous instance.

**Pull request merged**